using_restricted_array.rb

```diff
@@ -60,7 +134,29 @@ def find_largest(array, length)
 # (Hint: if there are elements with 'SPECIAL_VALUE', there is no room to insert)
 # All subsequent elements will need to be moved forward by one index.
 def insert_ascending(array, length, value_to_insert)
-  puts "NOT IMPLEMENTED"
+  i = 0
+
+  until array[i] > value_to_insert
+    if array[i] == SPECIAL_VALUE
+      return

```

Comment:
In this case, you should be inserting at array[i] and replace SPECIAL_VALUE with value_to_insert.